src/Units/UnitModel.java

```diff
@@ -80,8 +84,8 @@ public static String standardizeID(String input)
   public long carryableExclusionMask;
   private int COstr = 100;
   private int COdef = 100;
-  public double COcost = 1.0;
-  public int moneyCostAdjustment = 0;
+  public double costMultiplier = 1.0;
+  public int costShift = 0;
```

Comment:
These don't appear to be used anymore. Should COstr/def (and maybe other things) also be eliminated in favor of using UnitMods so that the model can be properly immutable?

Reply:
They are used, specifically as the basis for `UnitContext`'s comparable fields, and as implementation hooks for CO D2Ds.
If we wanted immutable `UnitModel`s and also D2Ds, we *could* throw in Lombok to get copy-on-write semantics running, I think.

I don't hate the idea of doing away with these fields entirely and running with `UnitModifier` only, though.

Reply:
Are we directly comparing `UnitContext`s anywhere? That seems like a bad idea, given how willy-nilly they are created.
I'm not gonna advocate for copy-on-write, but managing D2D/all buffs with `UnitModifiers` seems fine, and is where I thought you were going (and you already have a unit modifier to change cost, so this seems redundant now).

Reply:
> Are we directly comparing `UnitContext`s anywhere? That seems like a bad idea, given how willy-nilly they are created.

I don't think so. I meant that changes to these fields in the `UnitModel` would propagate to any `UnitContext`s that were created from it.
> managing D2D/all buffs with `UnitModifier`s seems fine, and is where I thought you were going

Huh. Okay. I don't mind. We wanna do that in this PR?

Reply:
I don't think I'm opposed. Like I said, I thought that was part of your intent, though to be fair I'm not really sure what the scope of this PR is even supposed to be.

Reply:
The scope was supposed to be "Introduce `UnitModifier` so we don't have to directly change `UnitModel`s when activating most abilities".
It didn't occur to me that that same concept would work just as well to implement most D2Ds.
I feel like this PR is big enough at this point that we want to make a new one for new projects like "fully-immutable, shared `UnitModel`s for all COs in a match".